File: frontend/packages/shatra-rules/src/domain.js
export function isOwnColor(pieceName, color) {
  if (!pieceName) return false;
  if (pieceName.includes('бел')) return color.startsWith('бел');
  return color.startsWith('чер');
}

export function parsePieceName(name) {
  if (!name) throw new Error('Empty piece name');
  const color = name.includes('бел') ? 'белый' : 'черный';
  let pieceType = 'батыр';
  if (name.includes('шатра')) pieceType = 'шатра';
  else if (name.includes('бий')) pieceType = 'бий';
  return { color, pieceType };
}

File: crates/shatra-engine/src/rules/domain.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum PieceType {
    Shatra,
    Biy,
    Batyr,
}

pub fn is_own_color(piece_name: &str, color: &str) -> bool {
    if piece_name.contains("бел") {
        return color.starts_with("бел");
    }
    color.starts_with("чер")
}

pub fn parse_piece_name(name: &str) -> Option<(String, PieceType)> {
    if name.is_empty() {
        return None;
    }
    let color = if name.contains("бел") {
        "белый".to_string()
    } else {
        "черный".to_string()
    };
    let piece_type = if name.contains("шатра") {
        PieceType::Shatra
    } else if name.contains("бий") {
        PieceType::Biy
    } else {
        PieceType::Batyr
    };
    Some((color, piece_type))
}

pub fn opponent(color: &str) -> String {
    if color == "белый" {
        "черный".to_string()
    } else {
        "белый".to_string()
    }
}
